### PR TITLE
add take

### PR DESCRIPTION
**Pull request by Rubtsowa**

### CODE REVIEW

**Review by densmirn — COMMENTED** [review]
Now we are able to run `numpy` tests under `dpnp` with the command `python -m tests_external.numpy.runtests`. So I found at the least one test that can be unskipped. It's `tests/test_numeric.py::TestNonarrayArgs::test_take` in the file `tests_external/skipped_tests_numpy.tbl`. So please remove the line from the file and check that the test is really passed.

**Comment by densmirn** [comment]
> Now we are able to run `numpy` tests under `dpnp` with the command `python -m tests_external.numpy.runtests`. So I found at the least one test that can be unskipped. It's `tests/test_numeric.py::TestNonarrayArgs::test_take` in the file `tests_external/skipped_tests_numpy.tbl`. So please remove the line from the file and check that the test is really passed.

You could run this test with the command:
```
python -m tests_external.numpy.runtests core/tests/test_numeric.py::TestNonarrayArgs::test_take
```

**Comment by densmirn** [comment]
> > Now we are able to run `numpy` tests under `dpnp` with the command `python -m tests_external.numpy.runtests`. So I found at the least one test that can be unskipped. It's `tests/test_numeric.py::TestNonarrayArgs::test_take` in the file `tests_external/skipped_tests_numpy.tbl`. So please remove the line from the file and check that the test is really passed.
> 
> You could run this test with the command:
> 
> ```
> python -m tests_external.numpy.runtests core/tests/test_numeric.py::TestNonarrayArgs::test_take
> ```

You can find the test here https://github.com/numpy/numpy/blob/d7a75e8e8fefc433cf6e5305807d5f3180954273/numpy/core/tests/test_numeric.py#L267-L273.

File: tests/test_indexing.py [hunk 1]
```diff
@@ -94,6 +94,37 @@ def test_put3():
...
+                          [[[[1, 2], [3, 4]], [[1, 2], [2, 1]]], [[[1, 3], [3, 1]], [[0, 1], [1, 3]]]],
+                          [[[[1, 2, 3], [3, 4, 5]], [[1, 2, 3], [2, 1, 0]]], [
+                              [[1, 3, 5], [3, 1, 0]], [[0, 1, 2], [1, 3, 4]]]],
+                          [[[[1, 2, 3], [4, 5, 6]], [[7, 8, 9], [10, 11, 12]]], [[[13, 14, 15], [16, 17, 18]], [[19, 20, 21], [22, 23, 24]]]]],
+                         ids=['[[0, 1, 2], [3, 4, 5], [6, 7, 8]]',
+                              '[[0, 1, 2, 3, 4], [5, 6, 7, 8, 9]]',
+                              '[[[1, 2], [3, 4]], [[1, 2], [2, 1]], [[1, 3], [3, 1]]]',
+                              '[[[[1, 2], [3, 4]], [[1, 2], [2, 1]]], [[[1, 3], [3, 1]], [[0, 1], [1, 3]]]]',
+                              '[[[[1, 2, 3], [3, 4, 5]], [[1, 2, 3], [2, 1, 0]]], [[[1, 3, 5], [3, 1, 0]], [[0, 1, 2], [1, 3, 4]]]]',
+                              '[[[[1, 2, 3], [4, 5, 6]], [[7, 8, 9], [10, 11, 12]]], [[[13, 14, 15], [16, 17, 18]], [[19, 20, 21], [22, 23, 24]]]]'])
```
Comment (samir-nasibli) [inline tests/test_indexing.py:117]: Can you, please, refactor this code?
Some tips

```python
INDICES = [[[0, 0], [0, 0]],
                    [[1, 2], [1, 2]],
                    [[1, 2], [3, 4]]]
ARRAY = [[[0, 1, 2], [3, 4, 5], [6, 7, 8]],
                          [[0, 1, 2, 3, 4], [5, 6, 7, 8, 9]],
                          [[[1, 2], [3, 4]], [[1, 2], [2, 1]], [[1, 3], [3, 1]]],
                          [[[[1, 2], [3, 4]], [[1, 2], [2, 1]]], [[[1, 3], [3, 1]], [[0, 1], [1, 3]]]],
                          [[[[1, 2, 3], [3, 4, 5]], [[1, 2, 3], [2, 1, 0]]], [
                              [[1, 3, 5], [3, 1, 0]], [[0, 1, 2], [1, 3, 4]]]],
                          [[[[1, 2, 3], [4, 5, 6]], [[7, 8, 9], [10, 11, 12]]], [[[13, 14, 15], [16, 17, 18]], [[19, 20, 21], [22, 23, 24]]]]],
@pytest.mark.parametrize("indices", INDICES, "ids", str(INDICES))
@pytest.mark.parametrize("array", ARRAY, "ids", str(ARRAY))

```

**Pull request merged**